schema + entity + sample test

E-Y-J opened the pull request:
implemented schema + entity and sample test using Data jpa test

E-Y-J committed:
schema + entity + sample test

implemented schema + entity and sample test using Data jpa test

krysseltillada commented on .settings/org.springframework.ide.eclipse.prefs:
```diff
@@ -0,0 +1,2 @@
+boot.validation.initialized=true
```
Please remove this, we don't need to keep & track such IDE-specific files inside the repo.

If someone, would clone this repo and uses eclipse to open this project, his or her IDE settings would get fucked up.

krysseltillada commented on src/main/java/libraryapp/domain/model/Book.java:
```diff
@@ -0,0 +1,141 @@
+	private String author;
+	
+	@Column(name = "date_published")
+	private Date date_published;
```
Please follow standard java naming conventions, usually, we follow camelCase namings for the methods and variables and PascalCase for the classes.

See https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

krysseltillada commented on src/main/java/libraryapp/domain/model/Book.java:
```diff
@@ -0,0 +1,141 @@
+
+	public Book(String title, String author, Date date_published, String category, String isbn10,
+			String isbn13) {
+		super();
```
Can be removed

krysseltillada commented on src/main/java/libraryapp/domain/model/Book.java:
```diff
@@ -0,0 +1,141 @@
+	private String category;
+	
+	@Column(name = "isbn10")
+	private String isbn10;
```
We could encapsulate these ISBNs into a separate class and use JPA's embeddable for this.

krysseltillada commented on src/main/java/libraryapp/domain/model/Book.java:
```diff
@@ -0,0 +1,141 @@
+	@Column(name = "isbn13")
+	private String isbn13;
+	
+	protected Book() {}
```
We need this since JPA requires us to create a constructor with protected being the minimum access level.

It's good that we have this, but unfortunately one will be wondering why this code exists.

Better have a comment that tells that this is required by the JPA specification.

krysseltillada commented on src/main/java/libraryapp/domain/model/Book.java:
```diff
@@ -0,0 +1,141 @@
+	}
+
+	public Long getId() {
+		return id;
```
Please check your IDE settings to force the generations for the getters to have the `this` keyword.

krysseltillada commented on src/main/java/libraryapp/domain/model/BookRepository.java:
```diff
@@ -0,0 +1,18 @@
+public interface BookRepository extends Repository<Book, Long> {
+
+	@Query("FROM Book a WHERE a.id = :id")
+	Book findById(@Param("id") long id);
```
We don't need such @Query here, a simple method derivation from spring data will suffice.

krysseltillada commented on target/classes/application.properties:
```diff
@@ -1,2 +1,10 @@
-logging.level.libraryapp=INFO
-logging.level.org.springframework=WARN
+# Control how Boot loads data when it starts:
+spring.jpa.hibernate.ddl-auto=none
```
Please don't push any IDE-specific files or files that result from a build.

krysseltillada commented on src/main/resources/data.sql:
```diff
@@ -0,0 +1,8 @@
+INSERT INTO book (title, author, date_published, category, isbn10, isbn13)
```
My first impression is that we're gonna use the bootstrapping code from the existing tests that we have. So we don't need this.

krysseltillada commented on src/test/java/libraryapp/model/BookRepositoryTests.java:
```diff
@@ -0,0 +1,51 @@
+import libraryapp.domain.model.BookRepository;
+
+@DataJpaTest
+public class BookRepositoryTests {
```
We don't need such tests that obviously works :)

krysseltillada commented on src/test/java/libraryapp/model/BookRepositoryTests.java:
```diff
@@ -0,0 +1,51 @@
+	private BookRepository bookRepository;
+	
+	@Test
+	void testFindOneBook() throws Exception {
```
It's obvious that this method is testing something, better remove those `test*` prefixes.

krysseltillada commented on .project:
```diff
@@ -15,6 +15,11 @@
 			</arguments>
 		</buildCommand>
+		<buildCommand>
+			<name>org.springframework.ide.eclipse.boot.validation.springbootbuilder</name>
```
What's the purpose of this?

krysseltillada commented on src/main/java/libraryapp/domain/model/LoanPeriod.java:
```diff
@@ -0,0 +1,105 @@
+	protected LoanPeriod() {}
+
+	public LoanPeriod(MemberCategory member_category, String book_category, Integer number_of_days) {
+		super();
```
We don't need this.

E-Y-J committed:
updated changes

added gitignore, changed membercategory into an entity, changed naming conventions, created isbn class, added this. to getters

E-Y-J committed:
gitignore

E-Y-J committed:
Delete .gitignore

E-Y-J committed:
Create .gitignore

E-Y-J committed:
updated schema, entities and mapping, and added 1 working springboottest

updated schema, entities and mapping, and added 1 working springboottest
